design-proposals/002-filter-data.md

```diff
@@ -0,0 +1,41 @@
+## Filter data in integration flow
+
+* Issue: https://github.com/syndesisio/syndesis-project/issues/2
+* Sprint: 12
+
+### Story
+
+As a citizen user, I would like to restrict the data that flows through an integration to match a specific set of criteria.  For example, I may have an integration that is triggered every time a tweet mentions my account, but I only want to process tweets that contain the text "Syndesis".
+
+I have two initial requirements for filtering data in an integration:
+* Based on content in the body.
+* Based on content in a message header.
+
+In both cases, I expect to be able to apply basic string operations to the data to form a filter expression (contains, does not contain, starts with).  For more details on options, see the Operator Support section here:
+http://camel.apache.org/simple.html
+
+Similar to logging, it would be nice if the UI provided help in creating filter expressions for new/inexperienced users (like through clickable examples).  The interface should also allow for advanced users to enter free form expressions.
```

Comment:
> it would be nice

I think we should be more prescriptive here. Does this proposal set out that this should happen (I think it should personally)? If so, then this should be stated positively.